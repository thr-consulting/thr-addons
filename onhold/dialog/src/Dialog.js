// @flow

import React, {Children} from 'react';
import type {ChildrenArray} from 'react';
import {Segment, Container} from 'semantic-ui-react';

type Props = {
	children: ChildrenArray<*>,
};

/**
 * A container component for custom slide up dialogs.
 * @class
 * @property {ChildrenArray} children - Dialog internal components
 */
export default function Dialog(props: Props) {
	if (Children.count(props.children) === 1) {
		return (
			<Segment basic>
				<Container>
					{props.children}
				</Container>
			</Segment>
		);
	}

	const children = Children.toArray(props.children);
	let buttons;
	if (children[children.length - 1].type.name === 'DialogButtons') {
		buttons = children.pop();
	}

	return (
		<Segment basic>
			<Container>
				{children}
			</Container>
			{buttons}
		</Segment>
	);
}
