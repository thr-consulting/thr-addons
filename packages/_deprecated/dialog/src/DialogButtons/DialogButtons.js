// @flow

import React from 'react';
import type {ChildrenArray} from 'react';
import {Divider, Button, Container} from 'semantic-ui-react';
import './dialogButtons.css';

type Props = {
	children: ChildrenArray<*>,
	onApprove: () => {},
	onReject: () => {},
	onRemove: () => {},
	container?: boolean,
};

/**
 * Standard set of buttons for use with Dialog.
 * @class
 * @property {ChildrenArray} children - Override with custom buttons if needed.
 * @property {function} onApprove - Called when the dialog is approved.
 * @property {function} onReject - Called when the dialog is canceled.
 * @property {function} onRemove - Called when the delete action is clicked.
 * @property {bool} [container=true] - If true, applies the Semantic UI container class to the buttons.
 */
export default function DialogButtons(props: Props) {
	const deleteButton = props.onRemove ? <Button onClick={props.onRemove}>Delete</Button> : null;
	const approveButton = props.onApprove
		?	<Button positive floated="right" onClick={props.onApprove}>Save</Button> : null;
	const rejectButton = props.onReject ? <Button color="black" onClick={props.onReject}>Cancel</Button> : null;
	const Wrapper = props.container === false ? 'div' : Container;

	const buttonContainer = props.children ? (
		<Wrapper className="dialog_buttonWrapper">
			{props.children}
		</Wrapper>
	) : (
		<Wrapper className="dialog_buttonWrapper">
			{deleteButton}
			<div className="dialog_rightButtons">
				{approveButton}
				{rejectButton}
			</div>
		</Wrapper>
	);

	return (
		<div>
			<Divider/>
			{buttonContainer}
		</div>
	);
}
