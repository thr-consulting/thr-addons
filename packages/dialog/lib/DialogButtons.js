// @flow

import React, {Component} from 'react';
import type {ChildrenArray} from 'react';
import {Divider, Button, Container} from 'semantic-ui-react';
import styles from './dialog.css';

type Props = {
	children: ChildrenArray<*>,
	onApprove: () => {},
	onReject: () => {},
	onRemove: () => {},
	container: boolean,
};

/**
 * Standard set of buttons for use with Dialog.
 * @class
 * @property {ChildrenArray} children - Override with custom buttons if needed.
 * @property {function} onApprove - Called when the dialog is approved.
 * @property {function} onReject - Called when the dialog is canceled.
 * @property {function} onRemove - Called when the delete action is clicked.
 * @property {bool} [container=true] - If true, applies the Semantic UI container class to the buttons.
 */
export default class DialogButtons extends Component<Props> {
	static defaultProps = {
		container: true,
	};

	props: Props;

	render() {
		const deleteButton = this.props.onRemove ? <Button onClick={this.props.onRemove}>Delete</Button> : null;
		const approveButton = this.props.onApprove ? <Button positive floated="right" onClick={this.props.onApprove}>Save</Button> : null;
		const rejectButton = this.props.onReject ? <Button color="black" onClick={this.props.onReject}>Cancel</Button> : null;
		const Wrapper = this.props.container ? Container : 'div';

		const buttonContainer = this.props.children ? (
			<Wrapper className={styles.buttonWrapper}>
				{this.props.children}
			</Wrapper>
		) : (
			<Wrapper className={styles.buttonWrapper}>
				{deleteButton}
				<div className={styles.rightButtons}>
					{approveButton}
					{rejectButton}
				</div>
			</Wrapper>
		);

		return (
			<div>
				<Divider/>
				{buttonContainer}
			</div>
		);
	}
}
