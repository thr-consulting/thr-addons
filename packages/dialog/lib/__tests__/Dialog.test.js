import React from 'react';
import {shallow} from 'enzyme';
import Dialog from '../Dialog';
import DialogButtons from '../DialogButtons';

describe('DialogButtons', () => {
	it('should render a single child without throwing an error', () => {
		expect(shallow(
			<Dialog>
				<div>Hello World</div>
			</Dialog>
		)).toMatchSnapshot();
	});

	it('should render multiple children without throwing an error', () => {
		expect(shallow(
			<Dialog>
				<div>Hello</div>
				<div>World</div>
			</Dialog>
		)).toMatchSnapshot();
	});

	it('should render multiple children with a DialogButtons instance without throwing an error', () => {
		expect(shallow(
			<Dialog>
				<div>Hello</div>
				<div>World</div>
				<DialogButtons/>
			</Dialog>
		)).toMatchSnapshot();
	});

	// it('should respond to changes', () => {
	// 	const mockOnChange = jest.fn();
	//
	// 	const wrapper = mount(
	// 		<FieldMap
	// 			value="initial value"
	// 			as={Input}
	// 			asValue="value"
	// 			onChange={mockOnChange}
	// 		/>);
	// 	const u = wrapper.find('input');
	// 	u.simulate('change', {target: {value: 'blah blah'}});
	//
	// 	expect(mockOnChange.mock.calls.length).toBe(1);
	// 	expect(mockOnChange.mock.calls[0][0]).toEqual('blah blah');
	// });
});
