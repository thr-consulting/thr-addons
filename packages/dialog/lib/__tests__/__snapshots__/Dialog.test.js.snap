// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogButtons should render a single child without throwing an error 1`] = `
<Segment
  basic={true}
>
  <Container>
    <div>
      Hello World
    </div>
  </Container>
</Segment>
`;

exports[`DialogButtons should render multiple children with a DialogButtons instance without throwing an error 1`] = `
<Segment
  basic={true}
>
  <Container>
    <div>
      Hello
    </div>
    <div>
      World
    </div>
  </Container>
  <DialogButtons
    container={true}
  />
</Segment>
`;

exports[`DialogButtons should render multiple children without throwing an error 1`] = `
<Segment
  basic={true}
>
  <Container>
    <div>
      Hello
    </div>
    <div>
      World
    </div>
  </Container>
</Segment>
`;
